src/app.py

```diff
@@ -2511,16 +2511,21 @@ def update_datasets_datastatus():
             if isinstance(dataset[prop], str) and \
                     len(dataset[prop]) >= 2 and \
                     dataset[prop][0] == "[" and dataset[prop][-1] == "]":
-                dataset[prop] = dataset[prop].replace("'", '"')
-                dataset[prop] = json.loads(dataset[prop])
-                if len(dataset[prop]) > 0:
-                    dataset[prop] = dataset[prop][0]
+                prop_as_list = string_helper.convert_str_literal(dataset[prop])
+                if len(prop_as_list) > 0:
```

Comment:
@DerekFurstPitt we should check the data type instance then treat it separately as list or still a string. This `len()` works on both list and string.

Reply:
In this particular section, we can treat it the same whether its a list or a string. Now that lists are allowed to be sent to the data ingest board (where there is now code to handle this) we don't need to handle strings and lists differently here. Code is there specifically to handle the case of empty lists and turning them into empty strings instead. It is ok to use the len operation whether or not the property is a string or a list. if its an empty list `""` then an empty string is returned. If its an empty list, its converted to `""`. If its either a non-empty string, or a non-empty list, its returned as is. That's why the type isn't checked here. It no longer matters. But the length check needs to be there to handle the specific case of empty lists.